packages/block-library/src/navigation/index.php

```diff
@@ -79,44 +79,9 @@ function block_core_navigation_sort_menu_items_by_parent_id( $menu_items ) {
  * @return array An array of parsed block data.
  */
 function block_core_navigation_parse_blocks_from_menu_items( $menu_items, $menu_items_by_parent_id ) {
-	if ( empty( $menu_items ) ) {
-		return array();
-	}
-
-	$blocks = array();
-
-	foreach ( $menu_items as $menu_item ) {
-		$class_name       = ! empty( $menu_item->classes ) ? implode( ' ', (array) $menu_item->classes ) : null;
-		$id               = ( null !== $menu_item->object_id && 'custom' !== $menu_item->object ) ? $menu_item->object_id : null;
-		$opens_in_new_tab = null !== $menu_item->target && '_blank' === $menu_item->target;
-		$rel              = ( null !== $menu_item->xfn && '' !== $menu_item->xfn ) ? $menu_item->xfn : null;
-		$kind             = null !== $menu_item->type ? str_replace( '_', '-', $menu_item->type ) : 'custom';
-
-		$block = array(
-			'blockName' => isset( $menu_items_by_parent_id[ $menu_item->ID ] ) ? 'core/navigation-submenu' : 'core/navigation-link',
-			'attrs'     => array(
-				'className'     => $class_name,
-				'description'   => $menu_item->description,
-				'id'            => $id,
-				'kind'          => $kind,
-				'label'         => $menu_item->title,
-				'opensInNewTab' => $opens_in_new_tab,
-				'rel'           => $rel,
-				'title'         => $menu_item->attr_title,
-				'type'          => $menu_item->object,
-				'url'           => $menu_item->url,
-			),
-		);
-
-		$block['innerBlocks']  = isset( $menu_items_by_parent_id[ $menu_item->ID ] )
-			? block_core_navigation_parse_blocks_from_menu_items( $menu_items_by_parent_id[ $menu_item->ID ], $menu_items_by_parent_id )
-			: array();
-		$block['innerContent'] = array_map( 'serialize_block', $block['innerBlocks'] );
-
-		$blocks[] = $block;
-	}
+	_deprecated_function( __FUNCTION__, '5.8.0', 'WP_Navigation_Gutenberg::parse_blocks_from_menu_items' );
 
-	return $blocks;
+	return WP_Navigation_Gutenberg::parse_blocks_from_menu_items( $menu_items, $menu_items_by_parent_id );
```

Comment:
We should leave all the functions that we deprecate intact so what people who use them don't get surprising results when we chanage things in `WP_Navigation_Gutenberg` in the future.